dev/bots/test/common.dart

```diff
@@ -31,3 +35,190 @@ Matcher throwsExceptionWith(String messageSubString) {
     ),
   );
 }
+
+/// A matcher that matches error messages specified in the given `fixture` [File].
+///
+/// This matcher allows analyzer tests to specify the expected error messages
+/// in the test fixture file, eliminating the need to hard code line numbers in
+/// the test.
+///
+/// The error messages must be printed using the [foundError] function. Each
+/// error must start with the path to the file where the error resides, line
+/// number (1-based instead of 0-based) of the error, and a short description,
+/// delimited by colons (`:`). In the test fixture one could add the following
+/// comment on the line that would produce the error, to tell matcher what to
+/// expect:
+/// `// ERROR: <error message without the leading path and line number>`.
+Matcher matchesErrorsInFile(File fixture, {List<String> endsWith = const <Never>[]}) =>
```

Comment:
Would it be doable to add some tests for this method? I know, its indirectly tested via the other tests.

Reply:
Add a very simple test: https://github.com/flutter/flutter/pull/161108/commits/4d89a8a30af93ecb5278f0e82326bc9ab609fac4